packages/nimble-components/src/table-column/mixins/placeholder.ts

```diff
@@ -0,0 +1,32 @@
+import { attr } from '@microsoft/fast-element';
+import type { TableColumn } from '../base';
+
+// Pick just the relevant properties the mixin depends on (typescript complains if the mixin declares private / protected base exports)
+// Because the 'placeholder' mixin doesn't depend on any properties of the TableColumn, there are no properties to pick.
+type TableColumnWithPlaceholder = Pick<TableColumn, never>;
+// prettier-ignore
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type TableColumnWithPlaceholderConstructor = abstract new (...args: any[]) => TableColumnWithPlaceholder;
+
+// As the returned class is internal to the function, we can't write a signature that uses is directly, so rely on inference
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function mixinColumnWithPlaceholderAPI<
+    TBase extends TableColumnWithPlaceholderConstructor
+>(base: TBase) {
+    /**
+     * The Mixin that provides a concrete column with the API to allow grouping
+     * by the values in that column.
+     */
+    abstract class ColumnWithPlaceholder extends base {
+        public placeholder?: string;
+
+        public abstract placeholderChanged(): void;
```

Comment:
I debated on doing that. I was torn on whether it was better to have every column implement `placeholderChanged()` and do whatever they needed or to force each column that uses this mixin to have a `updateColumnConfig()` function. I landed on it being better to have columns implement `placeholderChanged()` with the flexibility of structuring the rest of their class however they wanted rather than require all columns with placeholders to implement updates through a `updateColumnConfig()` function.

I'm willing to be convinced otherwise.